pkg/controller/networkpolicy/networkpolicy_controller_perf_test.go

```diff
@@ -18,6 +18,7 @@ package networkpolicy
 
```

Comment:
perhaps unrelated to this PR, but I noticed a long time ago that we don't have any benchmark for policies using `NamespaceSelector` to select network peers. Could we add one?

Reply:
will add one

Reply:
Thanks for bring this up. I added a new case and found the new implementation made it worse for execution time. The reason was in previous code we got all namespaces that match namespaceSelector first then scan Pods in them. The new code scans all labels in all namespaces in this case so it became worse. I used the same way as previous code to optimize it, now it's better than previous code.